File: app/src/main/java/org/inventivetalent/notificationlogger/database/NotificationDao.java
package org.inventivetalent.notificationlogger.database;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.Date;
import java.util.List;

@Dao
public interface NotificationDao {

    @Query("SELECT * FROM notifications ORDER BY time DESC")
    List<Notification> getAll();

    @Query("SELECT * FROM notifications WHERE time > :since ORDER BY time DESC")
    List<Notification> getAllSince(Date since);


    @Insert
    void insert(Notification... notifications);

}

File: app/src/main/java/org/inventivetalent/notificationlogger/database/NotificationDao.kt
package org.inventivetalent.notificationlogger.database

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import java.util.*

@Dao
interface NotificationDao {

    @Query("SELECT * FROM notifications ORDER BY time DESC")
    fun getAll(): LiveData<List<Notification>>

    @Query("SELECT * FROM notifications WHERE time > :since ORDER BY time DESC")
    fun getAllSince(since: Date): LiveData<List<Notification>>


    @Insert
    suspend fun insert(vararg notifications: Notification)

}
